doc/releases.md

```diff
@@ -186,6 +186,12 @@ informed perspective, such as a member of the NAN team.
 see a need to bump `NODE_MODULE_VERSION` then you should consult the TSC.
 Commits may need to be reverted or a major version bump may need to happen.
 
+*Note*: The Node.js ecosystem is reliant on ABI compatibility within a Semver Major
+release. To maintain ABI compatibility it is expected that production builds of
+Node.js will be built against the same version of dependencies as the project
+vendors. If Node.js is to be built against a different version of a dependency
+please create a custom `NODE_MODULE_VERSION` to ensure ecosystem compatibility.
```

Comment:
We should probably strongly encourage people who do this to contact us, so that we have consistent `NODE_MODULE_VERSION` values between distributors?

Reply:
So qq, I wasn't sure the answer to, can we support NODE_MODULE_VERSION to have characters in it or does gyp assume it is a number?

Reply:
I guess the implication of this is we'll need to maintain a list of `NODE_MODULE_VERSION`. Do we have anything other than https://github.com/nodejs/node/blob/dcfd323c6be53c6ea812e5d98eee8f4e8d31dc57/src/node_version.h#L74-L114 ?